src/main/resources/category/java/common.xml

```diff
@@ -930,6 +930,192 @@ public RetrieveCache(final @Value("${cache.expiryTimeMillis}") long timeToLiveMi
+    and ((ancestor::ClassBody[1]//MethodDeclaration[(@Visibility='public' or ../../@Nested=true()) and @Static=false() and @Name='toString']
+            and count(ancestor::ClassBody[1]/FieldDeclaration[@Static=false()]) <=
+                (1 + count(ancestor::ClassBody[1]/MethodDeclaration[(@Visibility='public' or ../../@Nested=true()) and @Static=false() and (starts-with(@Name, 'get') and string-length(@Name) > 3 or starts-with(@Name, 'is') and string-length(@Name) > 2)]))
+            or ancestor::ClassDeclaration[1][ends-with(@SimpleName, 'Dto')])
+        (: or #fields == #getters :)
+        or count(ancestor::ClassBody[1]/FieldDeclaration[@Static=false()]) = count(ancestor::ClassBody[1]/MethodDeclaration[(@Visibility='public' or ../../@Nested=true()) and @Static=false() and (starts-with(@Name, 'get') and string-length(@Name) > 3 or starts-with(@Name, 'is') and string-length(@Name) > 2)])
+        or ancestor::ClassDeclaration[1][pmd-java:hasAnnotation('lombok.Getter')]
+        )
+] (:-up to class -:)
+]/..
+]]>
+                </value>
+            </property>
+        </properties>
+        <example>
+            <![CDATA[
+class Getters { // bad
+    private String someState1 = "some1";
+    private String someState2 = "some2";
+
+    public String getSomeState1() {
+        return someState1;
+    }
+    public String getSomeState2() {
+        return someState2;
+    }
+}
+
+@Getter
+class LombokGetterBad { // bad
+    private String someState1 = "some1";
+    private String someState2 = "some2";
+}
+
+@Getter
+@EqualsAndHashCode
+class LombokGetterGood {
+    private String someState1 = "some1";
+    private String someState2 = "some2";
+}
+            ]]>
+        </example>
+    </rule>
+
+    <rule name="EqualsOperationInconsistentWithHashCode"
+          class="net.sourceforge.pmd.lang.rule.xpath.XPathRule"
+          language="java"
+          message="Equals uses a conversion inconsistent with hashCode."
+          externalInfoUrl="${doc_root}/JavaCodeQuality.md#ieah03">
+        <description>Problem: Equal objects may have different hashCodes and end-up in different buckets of a Map/Set. Strange things can happen like adding an object to a Set and not being able to find it back.
+            Solution: When objects are equal, hashCode needs to be equal, too. Use the same fields in equals and hashCode and use identical conversions like toUpperCase() in both when needed. Don't use equalsIgnoreCase. (jpinpoint-rules)
+        </description>
+        <priority>1</priority>
+        <properties>
+            <property name="tags" value="jpinpoint-rule,correctness" type="String" description="classification"/>
+            <property name="xpath">
+                <value><![CDATA[
+//FieldDeclaration//VariableId[
+(@Name = ancestor::ClassBody[1]//MethodDeclaration[@Name='equals' and (FormalParameters/@Size = 1)]/Block//MethodCall[pmd-java:matchesSig('java.lang.String#toLowerCase()')]//VariableAccess/@Name) >
```

Comment:
can combine with next one by using `java:matchesSig('java.lang.String#toLowerCase()') or java:matchesSig('java.lang.String#toUpperCase()')`?

Reply:
Ah, I see now that it needs the `>` check for both separately. Clarify with comment?